src/search.c

```diff
@@ -67,15 +67,17 @@ setup_and_find_next(struct view *view, enum request request)
 	    && !utf8_string_contains_uppercase(view->env->search))
 		regex_flags |= REG_ICASE;
 
-	if (view->regex) {
-		regfree(view->regex);
-		*view->grep = 0;
-	} else {
+	if (!view->regex) {
 		view->regex = calloc(1, sizeof(*view->regex));
 		if (!view->regex)
 			return ERROR_OUT_OF_MEMORY;
 	}
 
+	if (*view->grep) {
+		regfree(view->regex);
+		*view->grep = 0;
```

Comment:
Yes, it is more readable to decouple the allocation from regular housekeeping needed when performing a new search.

Reply:
Yes and also, maybe I should have mentioned it, PCRE has a slight tendency to segfault when using `regfree()` on `view->regex` before `regcomp()`.